test/selenium/sessions_ui_test.js

```diff
@@ -47,115 +54,69 @@ describe('sessions ui', function() {
+              assert.equal(text, errorPasswordDoNotMatch);
+          }).then(function(){
+              return setUpPage.dismissAlert();
           });
-        });
       });
 
       it('should reject short passwords', function () {
-        return elements.pwd1.sendKeys('asdf').then(function() {
-          return elements.pwd2.sendKeys('asdf');
-        }).then(function() {
-          return elements.set.click();
-        }).then(function() {
-          return driver.wait(webdriver.until.alertIsPresent(), 5000);
-        }).then(function() {
-          return driver.switchTo().alert();
-        }).then(function(alert) {
-          return alert.getText().then(function(text) {
-            assert.equal(text,
-                         'Please use a password of at least 8 characters.');
+          return setUpPage.typePassword('asdf').then(function() {
+            return setUpPage.confirmTypePassword('asdf');
           }).then(function() {
-            alert.dismiss();
+            return setUpPage.tapSubmitButton();
+          }).then(function() {
+            return setUpPage.alertMessage();
+          }).then(function(text){
+              assert.equal(text, shortPasswordErrorMessage);
+          }).then(function() {
+              return setUpPage.dismissAlert();
           });
         });
-      });
 
       it('should fail if password is not set', function() {
-        return  elements.set.click().then(function() {
-          return driver.wait(webdriver.until.alertIsPresent(), 5000);
-        }).then(function() {
-          return driver.switchTo().alert();
-        }).then(function(alert) {
-          return alert.getText().then(function(text) {
-            assert.equal(text,
-                         'Please use a password of at least 8 characters.');
-          }).then(function() {
-            alert.dismiss();
+          return setUpPage.tapSubmitButton().then(function(){
+              return setUpPage.alertMessage();
+          }).then(function(text){
+              assert.equal(text, shortPasswordErrorMessage);
+          }).then(function(){
+              return setUpPage.dismissAlert();
           });
         });
-      });
 
-      it('should accept matching, long-enough passwords', function () {
-        return elements.pwd1.sendKeys(PASS).then(function() {
-          return elements.pwd2.sendKeys(PASS);
-        }).then(function() {
-          return elements.set.click();
-        }).then(function() {
-          return driver.findElement(webdriver.By.id('thank-you'));
-        }).then(function(elt) {
-          return driver.wait(webdriver.until.elementIsVisible(elt), 5000)
-            .then(function() {
-              return elt.getAttribute('innerHTML');
-            }).then(function(value) {
-              assert.equal(value, 'Thank you!');
-            });
-        });
+      it('should accept matching, long-enough passwords', function() {
+          return setUpPage.typePassword('12345678').then(function() {
```

Comment:
*Proposed improvement for a follow up*

Like in [this comment](https://github.com/fxbox/foxbox/pull/170/files#r55529272), we could factorize the sign up procedure in a function like `signUpSuccess(password)`